lms/djangoapps/student_account/views.py

```diff
@@ -74,6 +74,30 @@ def login_and_registration_form(request, initial_mode="login"):
     if UserProfile.has_registered(request.user):
         return redirect(redirect_to)
 
+    if third_party_auth.is_enabled():
+        force_provider_id = settings.FEATURES.get('FORCE_TPA')
+        if force_provider_id:
+            force_provider = third_party_auth.provider.Registry.get(
+                provider_id=force_provider_id
+            )
+            if force_provider and force_provider.display_for_login:
+                running_pipeline = third_party_auth.pipeline.get(request)
+                if not running_pipeline:
+                    if initial_mode == "login":
+                        tpa_login = pipeline.get_login_url(
+                            force_provider_id,
+                            pipeline.AUTH_ENTRY_LOGIN,
+                            redirect_url=redirect_to,
+                        )
+                        return redirect(tpa_login)
+                    elif initial_mode == "register":
+                        tpa_register = pipeline.get_login_url(
+                            force_provider_id,
+                            pipeline.AUTH_ENTRY_REGISTER,
+                            redirect_url=redirect_to,
+                        ),
```

Comment:
We have an errant comma here.
One of the classic arguments for sharing common code (like the redirect above) is that reduces the chances of typos like this :)

Reply:
Ah yea this comma cost me a while to debug last Friday but I must have forgot to make the fix on my local code as well; in any case, I'll refactor this block like you suggested